### PR TITLE
agent: return req error if prometheus metrics are disabled.

### LINKED ISSUE #10070: telemetry: Prometheus config flag `prometheus_metrics` doesn't alter endpoint behaviour
### Nomad version
`Nomad v1.0.3`

### Issue
When the `telemetry.prometheus_metrics` config value is set to false, calling `/v1/metrics?format=prometheus` still responds 200 OK and with Prometheus formatted data. In accordance with the [documentation](https://www.nomadproject.io/docs/configuration/telemetry#prometheus), I would expect this call to fail. 

### Reproduction steps

Config file:
```hcl
telemetry {
  prometheus_metrics = false
}
```

- run the Nomad agent using `nomad agent -dev -config=<FILE>`
- attempt to use the metrics endpoint using the Prometheus format `curl -sD - -o /dev/null 'localhost:4646/v1/metrics?format=prometheus'` viewing the response headers
- optionally view the response body from the endpoint `curl -s 'localhost:4646/v1/metrics?format=prometheus'`

### PR DESCRIPTION
**Pull request by jrasell**
If the user has disabled Prometheus metrics and a request is
sent to the metrics endpoint requesting Prometheus formatted
metrics, then the request should fail.

The returned response code copies that of Consul, to provide a
consistent user experience.

It also fixes a knock-on test failure which assumed Prometheus
formatted metrics were available from the test server.

closes #10070

### CODE REVIEW

**Review by cgbaker — APPROVED** [review]
LGTM 👍

File: command/agent/metrics_endpoint.go [hunk 1]
```diff
@@ -22,6 +22,12 @@ func (s *HTTPServer) MetricsRequest(resp http.ResponseWriter, req *http.Request)
 	}
 
 	if format := req.URL.Query().Get("format"); format == "prometheus" {
+
+		// Only return Prometheus formatted metrics if the user has enabled
+		// this functionality. GH-10070.
```
Comment (tgross) [inline command/agent/metrics_endpoint.go:27]: Usually we only leave GitHub issue IDs in if we _haven't_ fixed something. For something we've already fixed, the commit log will tell us what's up.

```suggestion
		// this functionality.
```

**Review by tgross — APPROVED** [review]
LGTM :+1:

Let's add a changelog entry and leave a note in the upgrades guide so that users notice this.

**Comment by jrasell** [comment]
requesting re-review to get eyes on the documentation changes I have added to this PR.

File: website/content/docs/upgrade/upgrade-specific.mdx [hunk 2]
```diff
@@ -14,6 +14,15 @@ upgrade. However, specific versions of Nomad may have more details provided for
 their upgrades as a result of new features or changed behavior. This page is
 used to document those details separately from the standard upgrade flow.
 
+## Nomad 1.1.0
+
+#### Agent Metrics API
+
+The Nomad agent metrics API now respects the
+[prometheus_metrics](https://www.nomadproject.io/docs/configuration/telemetry#prometheus_metrics)
```
Comment (tgross) [inline website/content/docs/upgrade/upgrade-specific.mdx:22]:
```suggestion
[`prometheus_metrics`](https://www.nomadproject.io/docs/configuration/telemetry#prometheus_metrics)
```

**Review by tgross — APPROVED** [review]
LGTM

**Review by cgbaker — APPROVED** [review]
LGTM, thanks, @jrasell

**Pull request merged**